### ISSUE TITLE
noArrayIndexKey: false positive for static lists + biome-ignore ineffective on multi-line JSX

### ISSUE THREAD

**Issue by mulkatz** [biomejs/biome#9469]
## Description

Two related issues with `lint/suspicious/noArrayIndexKey`:

### 1. False positive for static/fixed-length lists

The rule flags array index keys even when the list is completely static (fixed length, never reordered). This is a common React pattern for skeleton/placeholder loading states:

```jsx
{Array.from({ length: 5 }, (_, i) => (
  <SkeletonItem key={i} />
))}
```

This is safe because:
- The array length is a literal constant
- Items are never reordered, added, or removed
- There is no meaningful stable key to use

ESLint's equivalent rule (`react/no-array-index-key`) has the same limitation, but the community consensus is that static lists are a valid exception. It would be great if Biome could detect `Array.from({ length: <literal> })` or similar patterns and suppress the warning.

### 2. `biome-ignore` suppression doesn't work with multi-line JSX elements

When trying to suppress the rule with `biome-ignore`, it only works when `key` is on the same line as the element tag:

**Works:**
```jsx
{Array.from({ length: 5 }, (_, i) => (
  // biome-ignore lint/suspicious/noArrayIndexKey: static skeleton
  <SkeletonItem key={i} />
))}
```

**Does NOT work:**
```jsx
{Array.from({ length: 5 }, (_, i) => (
  // biome-ignore lint/suspicious/noArrayIndexKey: static skeleton
  <div
    key={i}
    className="..."
  >
    ...
  </div>
))}
```

In the second case, Biome reports:
- `suppressions/unused`: "Suppression comment has no effect"
- `lint/suspicious/noArrayIndexKey`: still fires on `key={i}`

It seems the suppression is matched to the line immediately following the comment, but the diagnostic is reported on the line where `key={i}` appears. When those are different lines (multi-line JSX), the suppression is ignored.

## Expected behavior

1. `noArrayIndexKey` should not flag index keys when the source array has a static/literal length
2. `biome-ignore` should suppress diagnostics on multi-line JSX elements when placed before the opening tag

## Environment

- Biome version: 2.4.6
- Framework: React 19 + TypeScript

**Comment by JuanICasareski** [comment]
+1 this. I think it could be an extra rule or maybe just a config, but I wouldn't make it a default.

I'd be willing to implement the following:
- ` Array.from({ length: 5 }, (_, i) => <Item key={i} />)`. detected as static, no warning
- `Array.from({ length: count }, (_, i) => <Item key={i} />)`. detected as dynamic, warning. same for `length: count()`
- `Array(5).fill(0).map((_, i) => <Item key={i} />)`. static, no warning
- `Array(n).fill(0).map((_, i) => <Item key={i} />)`. dynamic, warning
- `[...Array(5)].map((_, i) => <Item key={i} />)`. static, no warning
- `[1, 2, 3].map((x, i) => <Item key={i}>{x}</Item>)`. static, no warning
